### ISSUE TITLE
Is it possible to write a customized inference function within emlearn

### ISSUE THREAD

**Issue by DylanWangWQF** [emlearn/emlearn#102]
Hi, I just read your 'getting started' tutorials. 
Most of the functions like `training()` and `predict()`  are implemented  using the `sklearn`. (e.g., `test_pred = cmodel.predict(test[feature_columns])`)
So, If it's possible to train a decision tree model within `emlearn`, save the model locally, and then implement the customized inference function which will be used for the following inference?
For example, I want to change the input feature values in one of the tree nodes during the inference. 
Basically, I need to control the process of the inference flexibly.

**Comment by jonnor** [comment]
Hi @DylanWangWQF. There is no specific support or hooks for anything like that in emlearn. And I do not think that I would add it either, unless there would be a very wide-spread usecase for it.
But as emlearn is an open source project, you are of course to take the code and modify it as you see fit for your particular needs. Or just to take aspects that you like, for example how it reads the node information from scikit-learn or how the C code generation works.

**Comment by DylanWangWQF** [comment]
@jonnor Thanks for your suggestions!

**Comment by jonnor** [comment]
You are welcome! Best of luck with your project. Do send me a link if you get something working (with or without emlearn), as I love research into custom inference for tree-based models!

**Comment by DylanWangWQF** [comment]
> You are welcome! Best of luck with your project. Do send me a link if you get something working (with or without emlearn), as I love research into custom inference for tree-based models!

Sure, I will!
